src/summarizedexperiment/dispatchers/combiners.py

```diff
@@ -0,0 +1,101 @@
+import pandas as pd
+from biocframe import BiocFrame
+from multipledispatch import dispatch
+
+__author__ = "keviny2"
+__copyright__ = "keviny2"
+__license__ = "MIT"
+
+
+def validate_inputs(ignore_names: bool, prefer_left: bool):
+    """Ensure boolean values are valid.
+
+    Args:
+        ignore_names (bool): if `True`, index names will be ignored when combining.
+        prefer_left (bool): if `True`, only null values in the left dataframe
+            will be populated by values from the right dataframe.
+    """
+    if ignore_names and prefer_left:
+        raise ValueError("`ignore_names` and `prefer_left` cannot both be `True`")
+
+
+def combine_dataframes(
+    left: pd.DataFrame, right: pd.DataFrame, ignore_names: bool, prefer_left: bool
+) -> pd.DataFrame:
+    """Combine dataframes with the appropriate method.
+
+    Args:
+        left (pd.DataFrame): a dataframe.
+        right (pd.DataFrame): a dataframe.
+        ignore_names (bool): if `True`, index names will be ignored when combining.
+        prefer_left (bool): if `True`, only null values in the left dataframe
+            will be populated by values from the right dataframe.
```

Comment:
this should really be called `ignore_na` or something like that

Reply:
I feel this function can be way more simpler, although i might be missing something here. 

```python
def combine_dataframes(
    *frames: pd.DataFrame,
    ignore_names: bool = False,
    axis: int = 0,
) -> pd.DataFrame:
    return pd.concat(frames, ignore_index=ignore_names, axis=axis)
```

e.g:

```python
df1 = pd.DataFrame(
    {
        "A": ["A0", "A1", "A2", "A3"],
        "B": ["B0", "B1", "B2", "B3"],
        "C": ["C0", "C1", "C2", "C3"],
        "D": ["D0", "D1", "D2", "D3"],
    },
    index=[0, 1, 2, 3],
)

df3 = pd.DataFrame(
    {
        "A": ["A8", "A9", "A10", "A11"],
        "B": ["B8", "B9", "B10", "B11"],
        "G": ["G8", "G9", "G10", "G11"],
        "H": ["H8", "H9", "H10", "H11"],
    },
    index=[0, 1, 6, 7],
)


frames = [df1, df3]

row = pd.concat(frames, ignore_index=True, axis=0)
row_ignore_index = pd.concat(frames, ignore_index=False, axis=0)

col = pd.concat(frames, ignore_index=True, axis=1)
col_ignore_index = pd.concat(frames, ignore_index=True, axis=1)
```

Reply:
may be i need an example of why we need to do `combine_first` and have the `prefer_left` param.

Reply:
if `useNames=False`, `combineCols` does not perform any concatenation on the rows. so if we call `df1.combineCols(df3, useNames=False)`, the new row names would simply be the row names of the 'first' dataframe - in our case, `[0, 1, 2, 3]`. the example you gave is kind of confusing (not sure why `ignore_index` is different for `row_ignore_index` and `col_ignore_index`), but regardless, pd.concat will always increase the number of rows, which isn't what we want when `useNames=False`. here's some R code using Bioconductor's SummarizedExperiment:

```
library(SummarizedExperiment)

se <- SummarizedExperiment(list(counts=matrix(rpois(50, 10), ncol=10)))
colData(se)$A <- 1L
rowData(se)$A <- 1
rownames(se) <- letters[1:5]

se2 <- SummarizedExperiment(list(counts=matrix(rpois(50, 10), ncol=10), 
                                 normalized=matrix(rnorm(50), ncol=10)))
colData(se2)$A <- 2L
colData(se2)$B <- 3
rowData(se2)$B <- "B"
rownames(se2) <- letters[3:7]

stuff <- combineCols(se, se2, use.names = FALSE)
```

Reply:
I believe this is already fixed in the #17. I added a couple more to keep the index of the first object.